pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml

```diff
@@ -133,22 +145,35 @@ spec:
                           format: int64
                           type: integer
                         volumeSize:
+                          allOf:
+                          - pattern: ^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$
+                          - pattern: ^((?:[1-9][0-9]{0,3}|[1-4][0-9]{4}|[5][0-8][0-9]{3}|59000)(G|Gi)|([0-9]||[0-9][0-7]|58)(T|Ti))$
```

Comment:
Discussing offline. I think we could potentially separate out expressions for G and T to be closer to their maximums

Reply:
We can't have multiple patterns here, without manually editing the crds. Kubebuilder does not support that